src/main/java/io/github/acm19/aws/interceptor/http/AwsRequestSigningApacheInterceptor.java

```diff
@@ -129,13 +136,28 @@ private static Map<String, List<String>> headerArrayToMap(final Header[] headers
         return headersMap;
     }
 
-    private static boolean skipHeader(final Header header) {
+    /**
+     * Ignores {@code Host} headers and {@code Content-Length} as long as it either
+     * contains the value {@code 0} or the service to be signed is Serverless
+     * OpenSearch. Details in AWS documentation below.
+     *
+     * AWS documentation:
+     * <pre><a href="https://docs.aws.amazon.com/opensearch-service/latest/developerguide/serverless-clients.html
+     * #serverless-signing">
+     *   Signing requests to OpenSearch Serverless
+     * </a></pre>
+     *
+     * @param header the header to evaluate
+     * @return {@code true} if header must be ignored {@code false} otherwise
+     */
+    private boolean skipHeader(Header header) {
+               // Strip for Content-Length: 0 and Serverless
         return (HTTP.CONTENT_LEN.equalsIgnoreCase(header.getName())
-                && "0".equals(header.getValue())) // Strip Content-Length: 0
+                && (("0".equals(header.getValue())) || isServerlessOpenSearch))
```

Comment:
Something doesn't add up here for the managed service. Maybe we didn't need `Content-Length` for it if we included `x-amz-content-sha256: UNSIGNED-PAYLOAD`?

Reply:
Not sure I follow you, if you don't want to sign the payload, why using the interceptor? :thinking:

Reply:
Signing headers?

With TLS signing more or less data (body vs. headers) shouldn't really matter, as long as something is signed the server can ensure that the payload came from who claimed to send it.